src/charm.py

```diff
@@ -487,13 +500,21 @@ def _workload_reset(self) -> StatusBase:
         except MySQLDataPurgeError:
             return BlockedStatus("Failed to purge data dir")
 
-        return ActiveStatus()
+        return ActiveStatus(self.active_status_message)
 
     def _get_primary_address_from_peers(self) -> str:
         """Retrieve primary address based on peer data."""
-        for unit in self._peers.units:
-            if self._peers.data[unit]["member-role"] == "primary":
-                return self._peers.data[unit]["instance-hostname"]
+        for unit in self.peers.units:
```

Comment:
i wonder why we have this private helper method: should we be using `_mysql.get_cluster_primary_address()`?